modules/Core/src/main/java/org/terasology/logic/inventory/InventoryAuthoritySystem.java

```diff
@@ -205,12 +206,15 @@ private boolean giveItemToSlots(EntityRef instigator, EntityRef entity, EntityRe
             InventoryUtils.adjustStackSize(entity, slot, itemAt.stackCount + count);
         }
 
-        if (emptySlotNo > -1) {
-            ItemComponent sourceItem = item.getComponent(ItemComponent.class);
-            sourceItem.stackCount = (byte) emptySlotCount;
-            item.saveComponent(sourceItem);
-
-            InventoryUtils.putItemIntoSlot(entity, item, emptySlotNo);
+        if (!emptySlots.isEmpty()) {
```

Comment:
This if condition looks wrong to me. If we put copies into the empty slots then the original item will remain without location and does not get destroyed.